### PR TITLE
feat: implement launch json editor

### PR DESCRIPTION
**Pull request by Ricbet**
### Types

- [x] 🎉 New Features

### <samp>🤖 Generated by Copilot at 3a8ce32</samp>

*  Add and use React JSON Schema Form library to render and validate JSON schemas in the debug launch view ([link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-4ab5bf726d0813cd862123b2515e4201cc0c9b67031365c01107c10a0daf6b4bR26-R28), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-f86d1cd15965214231cce3fa2b44d0fa407e9832d41b5ba1afb1847ddabe6ebbR1-R23), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-22e325d5312d9c44550fad5f1619256919014d486f96e39dcb76c98c1ac799f3R1-R36), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-86ff74b03299d60f6ecd1dff7353a7b788b4cf82b31a23b34121a6db5b32b00bR1-R3), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-86ff74b03299d60f6ecd1dff7353a7b788b4cf82b31a23b34121a6db5b32b00bL15-R48), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-86ff74b03299d60f6ecd1dff7353a7b788b4cf82b31a23b34121a6db5b32b00bL40-R80), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-86ff74b03299d60f6ecd1dff7353a7b788b4cf82b31a23b34121a6db5b32b00bL53-R97), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-86ff74b03299d60f6ecd1dff7353a7b788b4cf82b31a23b34121a6db5b32b00bL73-R113), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-86ff74b03299d60f6ecd1dff7353a7b788b4cf82b31a23b34121a6db5b32b00bL155-R288), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-bae5761df273439b8c38dedcf4a6d84682f842eb74cc140b900b7be95694a716R1-R96), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-2cab35c675650588fe515cc4a0a677a3d682c5ab97db765762ac68b23c75ad60R1-R96), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-44a1e5b7f9df9d26b77afac1161fdfd7150ea048277ed91f0a55763512916967R1-R38), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-6e7f7e9f5e5e44e10732f9da7583ce43aa515c3a938f4b7fdd379e2dc7601c9bR1-R22), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-483d2e76d3cdfee74cfb9ce43e708f0814fd6e27ab8b431fbd6414944c69105dR1-R40), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-0b47ea19c9728a74dab1d95d912bbfb2059e790bb2eeb5d49963010a73ee4b2dR1-R65), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-293c8af0ffda57fcd1e7c10e376a2886b6299fbe7d91deb179215731ea40d3deR1-R70), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-f66d683d6cd38ba54412c1c0a107842d6d65eb4312f6980a82ad912c4d51d9f4R380-R381), [link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-a5b8f7c1565e0628de0e7e6586f3ec3457c5e82522354bbd3f58926ecd3cdeb7R357-R358))
* Add a new utility function `acquireAjv` to lazily import and return an instance of the `Ajv` validator library ([link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-9fd2241c22e0044e8474facd1ea57dcf10d55f0e7aeceb921f14b733fdf4a77dR1-R11))
* Simplify the signature of two methods in the `LaunchResourceProvider` class by removing unused parameters ([link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-469d2164fd977fac513d47f5fdae2bc3ec04d5896e3f576b075504a9c0e981afL41-R44))
* Remove a comment that is no longer relevant to the `LaunchResourceProvider` class ([link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-469d2164fd977fac513d47f5fdae2bc3ec04d5896e3f576b075504a9c0e981afL32))
* Add a new CSS class `.launch_schema_body_container` to style the container element of the launch schema body ([link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-1c9f4efbfc56bf5e861ebb6f10ddd4b5e01574289aceee3140b103097513ca47R62-R67))
* Fix a typo in the interface `IBreadCrumbPart` in the `editor` package ([link](https://github.com/opensumi/core/pull/2598/files?diff=unified&w=0#diff-4094653eb8c5e5971274e17d7a7d8a8837b1c4e2acd4804c1ef92a2b0b8178c6L421-R421))

dark 主题
image

light 主题
image

- [x] 支持 input 输入框来编辑 type 是 string 的 json schema
- [x] 支持以多项列表选择器来编辑 type 为 array 的 json schema
- [x] 支持以 select 选择器来编辑 type 为 object 的 json schema
- [x] 样式优化
- [x] 支持通过 debug 的 snippet items 来默认填充数据

### <samp>🤖 Generated by Copilot at 3a8ce32</samp>

This pull request introduces a new UI for the debug launch view using the React JSON Schema Form library and custom components. It also refactors and fixes some code in the `debug` and `editor` packages, and adds some localization and styling.

### CODE REVIEW

File: packages/debug/src/browser/preferences/launch.view.tsx [hunk 1]
```diff
@@ -1,3 +1,6 @@
+import { withTheme } from '@rjsf/core';
+import { RJSFSchema, StrictRJSFSchema } from '@rjsf/utils';
+import validator from '@rjsf/validator-ajv8';
```
Comment (erha19) [inline packages/debug/src/browser/preferences/launch.view.tsx:3]: 这个库的功能感觉跟 ajv 是重复的?

File: packages/debug/src/browser/preferences/launch.view.tsx [hunk 2]
```diff
@@ -1,3 +1,6 @@
+import { withTheme } from '@rjsf/core';
```
Comment (erha19) [inline packages/debug/src/browser/preferences/launch.view.tsx:1]: @rjsf 的库引入了 2.17+1.8 = 3.97MB 的包体积 （未压缩大小），相对的 @opensumi/ide-core-browser 只有 3.75MB （未压缩大小），看界面不是很复杂，是不是可以考虑自己做一下界面

**Review by erha19 — CHANGES_REQUESTED** [review]
另外好像少了返回源文件的能力？

**Comment by Ricbet** [comment]
> 另外好像少了返回源文件的能力？

这个 pr 不做

File: packages/debug/src/browser/preferences/launch.view.tsx (hunk 2 above)
Comment (Ricbet) [reply packages/debug/src/browser/preferences/launch.view.tsx:1]: 很复杂，不考虑自己做。

包的问题可以再优化优化

Comment (Ricbet) [reply packages/debug/src/browser/preferences/launch.view.tsx:1]: 不对呀，看了下 production min 文件也才 56K，在可接受范围

File: packages/debug/src/browser/preferences/templates/json-templates.module.less [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    font-size: 14px;
+    display: inline-block;
+  }
+}
+
+.array_field_template {
+  display: flex;
+  flex-direction: column;
+
+  .array_item_label {
```
Comment (erha19) [inline packages/debug/src/browser/preferences/templates/json-templates.module.less:34]: 空样式

File: packages/debug/src/browser/preferences/templates/json-templates.module.less [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  }
+
+  .array_item_description {
+    margin-bottom: 8px;
+  }
+
+  .array_item_add {
```
Comment (erha19) [inline packages/debug/src/browser/preferences/templates/json-templates.module.less:41]: 空样式

File: packages/debug/src/browser/preferences/launch.view.tsx (hunk 1 above)
Comment (Ricbet) [reply packages/debug/src/browser/preferences/launch.view.tsx:3]: 确实

File: packages/debug/src/browser/preferences/launch.view.tsx (hunk 2 above)
Comment (erha19) [reply packages/debug/src/browser/preferences/launch.view.tsx:1]: 从这里看的 https://www.npmjs.com/package/@rjsf/core，不一定准确

**Review by erha19 — CHANGES_REQUESTED** [review]

File: packages/debug/src/browser/preferences/launch.view.tsx (hunk 2 above)
Comment (erha19) [reply packages/debug/src/browser/preferences/launch.view.tsx:1]: 这块如果成本不高的话，后面设置面板也可以通过这个方案改造一下，提升一下整体可用性

Comment (Ricbet) [reply packages/debug/src/browser/preferences/launch.view.tsx:1]: 他是 monorepo 的，只导入 core 不会有那么大

File: packages/debug/src/browser/preferences/launch.view.tsx (hunk 1 above)
Comment (Ricbet) [reply packages/debug/src/browser/preferences/launch.view.tsx:3]: 这里还不太一样，我只用了 ajv 的 validate 方法，目的是从 oneOf 列表中找出符合验证条件的那一个 oneOf
image

而 `@rjsf/validator-ajv8` 是专门用来给 rjsf 做一些类似表单验证的东西，虽然也有内置了一个 isValid 方法，但他性能非常差，主要原因是源码内部使用了 ajv 的 compile，这是个非常耗性能的操作

image

image

**Review by erha19 — APPROVED** [review]

**Pull request merged**